tools/bapdoc.ml

```diff
@@ -0,0 +1,232 @@
+    "bap-demangle", "Bap_demangle.Std", "writing name demanglers";
+    "bap-dwarf", "Bap_dwarf.Std", "a native DWARF parser";
+    "bap-ida", "Bap_ida.Std","an interface to IDA Pro";
+    "bap-llvm", "Bap_llvm.Std", "an inteface to LLVM disassemblers and loaders";
+    "bap-plugins", "Bap_plugins.Std", "loading plugins";
+    "bap-recipe", "Bap_recipe", "loading recipes (packs of command line arguments)";
+    "bap-strings", "Bap_strings.Std", "various text utilities";
+    "bap-traces", "Bap_traces.Std", "working with execution traces";
+    "text-tags", "Text_tags", "Extension of Format's semantic tags";
+  ];
+]
+
+let frontends = [
+  "bap", "bap main frontend";
+  "bap-byteweight", "create, obtain and evaluate byteweight signatures";
+]
+
+let introduction = {|
+{2 Introduction}
+
+The Carnegie Mellon University Binary Analysis Platform (CMU BAP) is a
+suite of utilities and libraries that enables analysis of programs in
+their machine representation. BAP is written in OCaml, relies on
+dynamically loaded plugins for extensibility, and is widely used for
+security analysis, program verification, and reverse engineering.
+
+The framework consists of a bunch of libraries, plugins and
+frontends. The libraries provide code reusability, plugins facilitate
+extensibility and frontends serve as entry points.
+
+Frontends come with comprehensive manuals,
+that can be accessed by using [--help] command line options, or via
+the [man] command, if the manpath is configured correctly.
+Finally, you can access a man page for a plugin using
+[--<PLUGIN>-help] command line option of a frontend, e.g., [bap --map-terms-help].
+
+|}
+
+let packages =
+  List.concat_map ~f:(fun (_,ps) -> List.map ~f:(fun (p,_,_) -> p) ps) libraries
+
+let mkdir path =
+  if not (Sys.file_exists path) then
+    Unix.mkdir path 0o770
+
+let run cmd =
+  let res = Sys.command cmd in
+  if res <> 0 then
+    failwith ("Command: '" ^ cmd ^ "' failed")
+
+let dedot name =
+  match String.split name ~on:'.' with
+  | name' :: _ when name' = "Bap" -> name
+  | name :: _ -> name
+  | _ -> name
+
+let render_entry (_,entry,desc) =
+  sprintf "- {{!%s}%s} - %s" entry (dedot entry) desc
+
+let render_section (name,entries) =
+  sprintf "{1 %s}\n%s" name
+    (String.concat ~sep:"\n" @@ List.map ~f:render_entry entries)
+
+type info = {
+  man : string -> string;
+  help : string -> string;
+}
+
+let program = {
+  man  = ident;
+  help = sprintf "%s --help=groff"
+}
+
+let plugin = {
+  man = sprintf "bap-plugin-%s";
+  help = sprintf "bap --%s-help=groff";
+}
+
+let man3_redirection lib =
+  mkdir "man3";
+  let redirection = sprintf {|
+    <!doctype html>
+    <html>
+    <head>
+    <meta http-equiv="refresh" content="0; url='../odoc/%s/index.html'" />
+    </head>
+    <body></body>
+    </html> |} lib in
+  let name = sprintf "man3/%s.3.html" lib in
+  Out_channel.write_all name ~data:redirection
+
+let build_manual {man; help} tool =
+  let repair_links x =
```

Comment:
this function is also misnamed, as it actually creates the manpage as the side-effect.